shell/platform/windows/flutter_windows.cc

```diff
@@ -292,10 +292,12 @@ void FlutterDesktopMessengerSetCallback(FlutterDesktopMessengerRef messenger,
                                         const char* channel,
                                         FlutterDesktopMessageCallback callback,
                                         void* user_data) {
-  flutter::FlutterDesktopMessenger::FromRef(messenger)
-      ->GetEngine()
-      ->message_dispatcher()
-      ->SetMessageCallback(channel, callback, user_data);
+  auto engine =
+      flutter::FlutterDesktopMessenger::FromRef(messenger)->GetEngine();
+  if (engine) {
+    engine->message_dispatcher()->SetMessageCallback(channel, callback,
+                                                     user_data);
+  }
```

Comment:
If we move forward with this approach, we should also update `FlutterDesktopMessengerSendWithReply` and `FlutterDesktopMessengerSendResponse`.

Reply:
I think there is a difference between setting the handler and sending a message.  I think the OP was trying to set the handler to `nullptr` so that they know the proper cleanup is happening instead of relying on the engine to do so.  They wanted to make that not an error.  I think it would be acceptable to make that not a crash, but to keep sending messages in the destructors a crash (we can improve the crash messaging with an assert if we don't already).